crates/model/src/signature.rs

```diff
@@ -22,12 +23,16 @@ impl Default for SigningScheme {
     }
 }
 
-#[derive(Eq, PartialEq, Clone, Copy, Debug, Deserialize, Serialize, Hash)]
+#[derive(Eq, PartialEq, Clone, Debug, Deserialize, Serialize, Hash)]
 #[serde(rename_all = "lowercase")]
 #[serde(tag = "signingScheme", content = "signature")]
 pub enum Signature {
     Eip712(EcdsaSignature),
     EthSign(EcdsaSignature),
+    /// Smart contract signatures as defined in [EIP-1271][eip1271].
+    ///
+    /// [eip1271]: https://eips.ethereum.org/EIPS/eip-1271
+    Eip1271(#[serde(with = "crate::bytes_hex")] Vec<u8>),
```

Comment:
```suggestion
    Eip1271 {
        owner: H160,
        #[serde(with = "crate::bytes_hex")]
        signature: Vec<u8>,
    },
```

Note that EIP-1271 signatures for orders are encoded as `abi.encodePacked(owner, signature)`. See <https://github.com/cowprotocol/contracts/blob/9926a44be2af40e338e415b25a5a8a7258ffb26e/src/contracts/mixins/GPv2Signing.sol#L261-L279> for more details.